### PR TITLE
add option list-dsn

### PR DESCRIPTION
**Pull request by fraoustin**
## Description

add option list-dsn : get all alias in aliasdsn section of config file

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by tsroten** [comment]
Great idea!!! This is super helpful!

A couple of thoughts:
- can we mask the passwords in the output?
- can we change “myclirc” in the output to the actual config file location? This could be overrided and might not be named myclirc

File: mycli/main.py [hunk 1]
```diff
@@ -964,6 +964,8 @@ def get_last_query(self):
 @click.option('-D', '--database', 'dbname', help='Database to use.')
 @click.option('-d', '--dsn', default='', envvar='DSN',
               help='Use DSN configured into the [alias_dsn] section of myclirc file.')
+@click.option('--list-dsn', 'list_dsn', is_flag=True,
+        help='list of DSN configured into the [alias_dsn] section of pgclirc file.')
```
Comment (tsroten) [inline mycli/main.py]: Looks like a copy/paste mixup - “pgclirc”

**Comment by meeuw** [comment]
This looks certainly useful, for me it would be nice (for https://github.com/meeuw/wrapc) to only output the dsn names. I'm using wrapc for bash autocompletion.

File: mycli/main.py [hunk 2]
```diff
@@ -1014,7 +1016,16 @@ def cli(database, user, host, port, socket, password, dbname,
                   defaults_file=defaults_file, login_path=login_path,
                   auto_vertical_output=auto_vertical_output, warn=warn,
                   myclirc=myclirc)
-
+    if list_dsn :
+        try:
+            for alias in mycli.config['alias_dsn']:
+                click.secho(alias + " : " + mycli.config['alias_dsn'][alias])
+            sys.exit(0)
+        except Exception as err:
```
Comment (meeuw) [inline mycli/main.py]: Could you please change `Exception` to the exception you're trying to catch? Now every exception in the try block will be reported as `Invalid DSN`. This `except Exception as e` is used in other places but there the exception is reported without translation (`click.secho(str(e), err=True, fg='red')`).

File: mycli/main.py (hunk 1 above)
Comment (fraoustin) [reply mycli/main.py]: exact, i change

File: mycli/main.py (hunk 2 above)
Comment (fraoustin) [reply mycli/main.py]: exact, i add keyerror

**Comment by meeuw** [comment]
Thanks @fraoustin!

**Pull request merged**